snapcraft/cli/remote.py

```diff
@@ -0,0 +1,252 @@
+        branch = "master"
+
+        # Send local data to the remote repository
+        echo.info("Sending data to remote builder...")
+        if git:
+            url, branch = _send_current_tree(provider, lp.user, build_id + "-git")
+        else:
+            url = _copy_and_send_tree(
+                provider, lp.user, remote_dir, build_id, project.info.name
+            )
+
+        # Create build recipe
+        # (delete any existing snap to remove leftovers from previous builds)
+        lp.delete_snap()
+        lp.create_snap(url, branch, archs)
+
+        targets = " for {}".format(", ".join(archs)) if archs else ""
+        echo.info(
+            "Building package{}. This may take some time to finish.".format(targets)
+        )
+
+        # Start building
+        req_number = lp.start_build(timeout=5, attempts=5)
+        echo.info(
+            "If interrupted, resume with: 'snapcraft remote-build --recover {}'".format(
+                req_number
+            )
+        )
+        _monitor_build(lp)
+
+
+def _monitor_build(lp: LaunchpadClient) -> None:
+    lp.monitor_build()
+    echo.info("Build complete.")
+    lp.delete_snap()
+
+
+def _send_current_tree(provider: str, user: str, build_id: str) -> Tuple[str, str]:
+    if not os.path.exists(".git"):
+        raise errors.NotGitRepositoryError
+    repo = Repo(".")
+    branch = repo.branch_name
+    if repo.is_dirty:
+        echo.warning(
+            "The following files have uncommitted changes that will not be used:"
+        )
+        echo.warning("\n".join(repo.uncommitted_files))
+    url = repo.push_remote(provider, user, branch, build_id)
+    return url, branch
+
+
+def _copy_and_send_tree(
+    provider: str, user: str, remote_dir: str, build_id: str, name: str
+) -> str:
+    if os.path.exists(".git"):
+        echo.warning(
+            "Git repository found, use option --git to build the current branch."
+        )
+    work_dir = os.path.join(remote_dir, build_id)
+    wt = Worktree(
+        ".",
+        work_dir,
+        ignore=[name + "_*.snap", "buildlog_*.txt*", "parts", "stage", "prime"],
+    )
+    url = wt.add_remote(provider, user, build_id)
+    wt.sync()
+    wt.push()
+    return url
+
+
+def _check_supported_archs(archs: List[str]) -> None:
+    unsupported_archs = []
+    for item in archs:
+        if item not in _SUPPORTED_ARCHS:
+            unsupported_archs.append(item)
+    if unsupported_archs:
+        raise errors.UnsupportedArchitectureError(archs=unsupported_archs)
+
+
+def _list_architectures(project, arch: str) -> List[str]:
+    arch_from_yaml = _project_architectures(project)
+    if arch_from_yaml:
+        if arch:
+            raise errors.ConflictingArchListError()
+        archs = arch_from_yaml
+    elif arch == "all":
+        archs = _SUPPORTED_ARCHS
+    else:
+        archs = arch.split(",") if arch else []
+
+    return archs
+
+
+def _project_architectures(project) -> List[str]:
```

Comment:
can you add an action to this, get seems appropriate, so _get_project_architectures

Reply:
Renamed as described.